bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -407,4 +480,54 @@ private boolean itemIsForThisBridge(Item item, @Nullable Map<String, Object> con
                 value.getClass(), item.getName());
         return (instance == 1);
     }
+
+    /**
+     * Check for any missing accessories.
+     *
+     * If there are, return true so we know to increment the config version. UNLESS
+     * we're configured to use dummy accessories, in which case backfill it with a dummy.
+     *
+     * @return if we need to increment the configuration version
+     */
+    private boolean checkMissingAccessories() {
+        List<String> toRemove = new ArrayList<>();
+        newDummies = false;
+        for (Map.Entry<String, String> accessory : knownAccessories.entrySet()) {
+            if (!accessoryRegistry.getAllAccessories().containsKey(accessory.getKey())) {
+                if (settings.useDummyAccessories) {
+                    newDummies = true;
+                    logger.debug("Creating dummy accessory for missing item {}.", accessory.getKey());
+                    accessoryRegistry.addRootAccessory(accessory.getKey(),
+                            new DummyHomekitAccessory(accessory.getKey(), accessory.getValue()));
+                } else {
+                    toRemove.add(accessory.getKey());
+                }
+            }
+        }
+
+        toRemove.forEach(k -> knownAccessories.remove(k));
+        return !toRemove.isEmpty();
+    }
+
+    private void checkForDummyAccessories() {
+        var dummyAccessories = accessoryRegistry.getAllAccessories().values().stream()
+                .filter(a -> a instanceof DummyHomekitAccessory).toArray(HomekitAccessory[]::new);
```

Comment:
would not be forEach better here and we print warning to each dummy accessory?
e.g.
```

accessoryRegistry.getAllAccessories().values().stream()
                .filter(a -> a instanceof DummyHomekitAccessory).forEach(dummyAccessory -> logger.warn(
                            "{} has been replaced with a dummy. See https://www.openhab.org/addons/integrations/homekit/#dummy-accessories for more information.",
                            dummyAccessory.getSerialNumber().get())
```